server/service/apple_mdm.go

```diff
@@ -2090,6 +2090,10 @@ func ReconcileProfiles(
 	for _, p := range hostProfiles {
 		go func(pp *fleet.MDMAppleBulkUpsertHostProfilePayload) {
 			var err error
+			// TODO(mna): I'm not sure this is correct? We pass multiple host UUIDs
+			// to Install/Remove profile, but we loop over each hostProfiles (so p
+			// represents a single host), and we send a single CommandUUID which is
+			// unique per host?
```

Comment:
@mna I think the end result looks correct (each host ends ups with the correct profiles installed/removed) but we're sending duplicated commands.

It's okay to have a single CommandUUID targeting multiple hosts (you might want to execute the exact same command in multiple hosts)

It's not okay that we're looping over `hostProfiles`, maybe we need yet another `map`?

Reply:
Yeah, I meant that we create a unique `CommandUUID` per host, and we bundle host uuids per profile, but even if 10 hosts share the same profile, we execute the command 10 times, each time targeting 10 hosts, and each time with a different CommandUUID which is associated with each one of the hosts.

I think we need to have the same `CommandUUID` for all hosts that will be part of the same command. Once we have that, we can loop over something that will call the command only once per profile ID (and it will be called for all hosts that have this profile, and all will have the same command uuid). Does that make sense to you?

Reply:
ah, sorry! so I basically repeated what you said but with other words 😅 

that makes sense to me!